src/factories/ProposalFactory.sol

```diff
@@ -163,4 +174,13 @@ contract ProposalFactory is IProposalFactory {
     function getProposalIDCounter() external view returns (uint) {
         return _proposalIdCounter;
     }
+
+    function _dependencyInjectionRequired(bytes memory dependencydata)
+        internal
+        pure
+        returns (bool)
+    {
+        (bool hasDependency,) = abi.decode(dependencydata, (bool, string[]));
+        return hasDependency;
```

Comment:
Cant you just look if dependencyData is not empty?
if you set dependencyData initially to bytes("") instead of actually doing the abi.encode(emptyStringarray) this should be possible no?

Reply:
if I only check `dependencyData` is not empty then something like `abi.encode(uint256)` would also pass. This ensures that the bytes data at least has a boolean as the first param.

Reply:
this was agreed upon in an internal call. marking this as resolved.